app/src/full/java/io/homeassistant/companion/android/settings/SettingsWearViewModel.kt

```diff
@@ -0,0 +1,152 @@
+package io.homeassistant.companion.android.settings
+
+import android.app.Application
+import android.net.Uri
+import android.util.Log
+import android.widget.Toast
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateMapOf
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.android.gms.wearable.CapabilityClient
+import com.google.android.gms.wearable.DataClient
+import com.google.android.gms.wearable.DataEvent
+import com.google.android.gms.wearable.DataEventBuffer
+import com.google.android.gms.wearable.DataItem
+import com.google.android.gms.wearable.DataMap
+import com.google.android.gms.wearable.DataMapItem
+import com.google.android.gms.wearable.PutDataMapRequest
+import com.google.android.gms.wearable.Wearable
+import io.homeassistant.companion.android.R
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.common.data.integration.IntegrationRepository
+import kotlinx.coroutines.launch
+
+class SettingsWearViewModel(application: Application) :
+    AndroidViewModel(application),
+    DataClient.OnDataChangedListener {
+
+    val app = application

```

Comment:
Shouldn't need to do this, AndroidViewModel has a getter for getting the application.

Reply:
So I tried to use `getApplication` but the call failed. I can try it again. Maybe I need to assign it first before using in the functions?

Reply:
Ok this was my bad, I had to set a top level variable in order to properly attain it